lib/libzfs_core/libzfs_core.c

```diff
@@ -940,3 +948,51 @@ lzc_destroy_bookmarks(nvlist_t *bmarks, nvlist_t **errlist)
 
 	return (error);
 }
+
+/*
+ * Performs key management functions
+ *
+ * crypto_cmd should be a value from zfs_ioc_crypto_cmd_t. If the command
+ * specifies to load or change a wrapping key, the key should be specified in
+ * the hidden_args nvlist so that it is not logged
+ */
+int
+lzc_load_key(const char *fsname, boolean_t noop, nvlist_t *hidden_args)
+{
+	int error;
+	nvlist_t *ioc_args = NULL;
+
+	if (hidden_args == NULL)
+		return (EINVAL);
+
+	ioc_args = fnvlist_alloc();
+	fnvlist_add_nvlist(ioc_args, ZPOOL_HIDDEN_ARGS, hidden_args);
+	if (noop)
+		fnvlist_add_boolean(ioc_args, "noop");
+	error = lzc_ioctl(ZFS_IOC_LOAD_KEY, fsname, ioc_args, NULL);
+	nvlist_free(ioc_args);
+
+	return (error);
+}
+
+int
+lzc_unload_key(const char *fsname)
+{
+	return (lzc_ioctl(ZFS_IOC_UNLOAD_KEY, fsname, NULL, NULL));
+}
+
+int
+lzc_change_key(const char *fsname, nvlist_t *args, nvlist_t *hidden_args)
```

Comment:
I didn't realize that was the case. The `args` in this case are the equivalent of `props` in `lzc_create()`. I just used a different name. That said, here there are less possibilities (keylocation, keyformat, pbkdf2iters, pbkdf2salt vs all properties). Would you like me to change this? I'm inclined to just rename `args` to `props` so that the code in libzfs_crypto.c can function the same for `lzc_create()` as it does for `lzc_change_key()`. Right now the properties list is done by `populate_create_encryption_params_nvlists()` in libzfs_crypto.c for both code paths.

For the `hidden_args`. I will just do as you have suggested.